jina/executors/__init__.py

```diff
@@ -527,7 +529,10 @@ def inject_config(
         tmp = fill_metas_with_defaults(raw_config)
         tmp['metas']['pea_id'] = pea_id
         tmp['metas']['read_only'] = read_only
-
+        if kwargs.get('metas'):
```

Comment:
I think then we need to add `metas` as Optional argument

Reply:
Shouldn't that be `if 'metas' in kwargs`?